src/extension/auth.js

```diff
@@ -37,59 +37,94 @@ export async function configureAuthAndCorsHeaders() {
+            }],
+          },
+          condition: {
+            regexFilter: `^https://${adminHost}/(config/${owner}.json|[a-z]+/${owner}/.*)`,
+            requestDomains: [adminHost],
+            requestMethods: ['get', 'post', 'delete'],
+            resourceTypes: ['xmlhttprequest'],
+          },
+        });
+
+        const corsFilters = [`^https://[0-9a-z-]+--[0-9a-z-]+--${owner}.aem.(live|page)/.*`];
+        const project = await getConfig('sync', `${owner}/${repo}`);
+        if (project) {
+          const { host, previewHost, liveHost } = project;
+
+          const additionalFilters = [host, previewHost, liveHost]
+            .filter(Boolean) // Filter out undefined or null values
+            .map((domain) => `^https://${domain}/.*`);
 
-      const corsFilters = [`^https://[0-9a-z-]+--[0-9a-z-]+--${owner}.aem.(live|page)/.*`];
-      const project = await getConfig('sync', `${owner}/${repo}`);
-      if (project) {
-        const { host, previewHost, liveHost } = project;
+          corsFilters.push(...additionalFilters);
+        }
 
-        const additionalFilters = [host, previewHost, liveHost]
-          .filter(Boolean) // Filter out undefined or null values
-          .map((domain) => `^https://${domain}/.*`);
+        const corsRules = corsFilters.map((regexFilter) => ({
+          id: getRandomId(),
+          priority: 1,
+          action: {
+            type: 'modifyHeaders',
+            responseHeaders: [{
+              header: 'Access-Control-Allow-Origin',
+              operation: 'set',
+              value: '*',
+            }],
+          },
+          condition: {
+            regexFilter,
+            initiatorDomains: ['tools.aem.live', 'labs.aem.live'],
+            requestMethods: ['get'],
+            resourceTypes: ['xmlhttprequest'],
+          },
+        }));
 
-        corsFilters.push(...additionalFilters);
+        rules.push(...corsRules);
       }
 
-      const corsRules = corsFilters.map((regexFilter) => ({
-        id: getRandomId(),
-        priority: 1,
-        action: {
-          type: 'modifyHeaders',
-          responseHeaders: [{
-            header: 'Access-Control-Allow-Origin',
-            operation: 'set',
-            value: '*',
-          }],
-        },
-        condition: {
-          regexFilter,
-          initiatorDomains: ['tools.aem.live', 'labs.aem.live'],
-          requestMethods: ['get'],
-          resourceTypes: ['xmlhttprequest'],
-        },
-      }));
+      if (siteToken) {
+        rules.push({
+          id: getRandomId(),
+          priority: 1,
+          action: {
+            type: 'modifyHeaders',
+            requestHeaders: [{
+              operation: 'set',
+              header: 'authorization',
+              value: `token ${siteToken}`,
+            }],
+          },
+          condition: {
+            regexFilter: `^https://[a-z0-9-]+--${repo}--${owner}.aem.(page|live)/.*`,
+            requestMethods: ['get', 'post'],
```

Comment:
Do we still need the `post`?

Reply:
Good question... probably not

Reply:
Yes, we'll need it for same-site RUM requests.